app/src/main/java/ch/epfl/sweng/rps/models/User.kt

```diff
@@ -4,33 +4,37 @@ package ch.epfl.sweng.rps.models
  *
  */
 data class User(
-    val username: String?,
+    val username: String? = "",
 
-    val uid: String,
+    val uid: String = "",
 
-    val gamesHistoryPrivacy: Privacy,
+    val gamesHistoryPrivacy: String = "",
 
-    val hasProfilePhoto: Boolean,
+    val hasProfilePhoto: Boolean = false,
 
-    val email: String?
+    val email: String? = "",
+
+    val matchesList: List<String>? = null
 ) {
```

Comment:
What is the reason behind initializing the attributes of User to defaults (like "", false or null)?

Reply:
They have to be initialized because firebase needs also a 0-parameter constructor in order to function correctly.